internal/model/registry.go

```diff
@@ -87,3 +87,15 @@ func ObjectFromProto(p proto.Message) (Object, error) {
 	// match any underlining resource on our model objects.
 	return nil, fmt.Errorf("cannot find type from Protobuf message %s", expectedFullName)
 }
+
+// OptionsForType returns the options for the provided Type, granted that options exist
+// for the given type. In the event no options exist, DefaultObjectOptions is returned.
+func OptionsForType(typ Type) ObjectOptions {
+	if obj, err := NewObject(typ); err == nil {
```

Comment:
Would normally test this, but it'd require adding in a dummy type to be added and this code is not written in a way to remove the global state that's currently going on. There's some refactoring that really should happen to avoid this global state.